java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Schema.java

```diff
@@ -79,6 +82,20 @@ public static Schema deserialize(ByteBuffer buffer) {
     return convertSchema(org.apache.arrow.flatbuf.Schema.getRootAsSchema(buffer));
   }
 
+  /** Returns the schema from the message.
+   *
+   * <p>The message must have the schema field set or an
+   * exception will be raised.
+   */
+  public static Schema deserializeFromMessage(ArrowBatchMessage buffer) {
```

Comment:
Not sure about this one either, what's the difference between this and `MessageSerializer.deserializeSchema()`?

Reply:
it appears it might not be needed.